android/src/main/java/com/reactnativecommunity/webview/RNCWebView.java

```diff
@@ -0,0 +1,93 @@
+package com.reactnativecommunity.webview;
+
+import android.webkit.WebView;
+import android.widget.FrameLayout;
+
+import androidx.annotation.Nullable;
+
+import com.facebook.common.logging.FLog;
+import com.facebook.react.uimanager.ThemedReactContext;
+
+public class RNCWebView extends FrameLayout {
```

Comment:
> It would be nice to have comments on this and InternalWebView to describe their unique roles, as that seems like it will be a source of confusion.
Sounds good.

> I can't help wondering if InternalWebView should be kept as RNCWebView (because it's the thing that actually extends WebView?) and this should be called something like RNCWebViewContainer or RNCWebViewProxy...?

`RNCWebViewContainer`/ `RNCWebViewProxy` does make sense, the main reason for this name change is to be consistent with the iOS counterpart. iOS `RNCWebView` is a wrapper over `WKWebView`

> The new-arch branch from @Titozzz extracts all the classes out from RNCWebViewManager into separate files... if InternalWebView were not a nested class of RNCWebViewManager, would it still be called Internal

Yeah, I called it `Internal` mainly to differentiate it from `RNCWebView`. Internal may not be the best name here, but let me know if you have any ideas.

If we are okay to have `RNCWebView` on Android to have a different meaning than the `iOS` counterpart, I think `RNCWebViewContainer` is perfectly fine

Reply:
Addressed here: https://github.com/react-native-webview/react-native-webview/pull/2469/commits/2eeb82f1377a4ae3f12cfb3ffad7b0f40e594f8c .

For now, I've renamed the outer WebView container to `RNCWebViewContainer`, and I've renamed the internal WebView (the one that extends the Android `WebView`) to `RNCWebView`.  This will make it slightly confusing for anyone working on both iOS and Android for this library, since now the iOS "RNCWebView" corresponds to the Android "RNCWebViewContainer" while the iOS "WKWebView" corresponds to the Android "RNCWebView". I'm okay with that if the inconsistency allows us to create better backward compatibility for the customization use case though.

Thanks for calling out that use case! Let me know if I missed anything